tests/integrations/celery/test_celery.py

```diff
@@ -722,3 +735,27 @@ def task(): ...
     (event,) = events
     (span,) = event["spans"]
     assert span["data"]["messaging.system"] == system
+
+
+@pytest.mark.forked
+@pytest.mark.parametrize("system", ("amqp", "redis"))
+def test_producer_span_data(
+    system, sentry_init, init_celery_producer_only, capture_events
+):
+    sentry_init(integrations=[CeleryIntegration()], enable_tracing=True)
+    celery = init_celery_producer_only(__name__, broker=f"{system}://example.com")
+    events = capture_events()
+
+    @celery.task()
+    def task(): ...
+
+    with start_transaction():
+        task.apply_async()
+
+    (event,) = events
+    span = next(span for span in event["spans"] if span["op"] == "queue.publish")
+    assert span["data"]["messaging.system"] == system
+
+    assert span["data"]["messaging.destination.name"] == "celery"
+    assert "messaging.message.id" in span["data"]
+    assert span["data"]["messaging.message.retry.count"] == 0
```

Comment:
I put all of these in a single test, rather than splitting into multiple tests like for the consumer, since it appears I need to fork them in order for my `init_celery_producer_only` fixture to work